### ISSUE TITLE
zfs snapshot does not deleted when deleting clone

### ISSUE THREAD

**Issue by redixin** [lxc/lxc#401]
# lxc-create -B zfs -n orig -t ubuntu
 ## ... skipped ...
 # lxc-clone -s orig snapshot
 could not find any snapshots to destroy; check snapshot names. # https://github.com/lxc/lxc/issues/213
 Created container snapshot as snapshot of orig
 # zfs list -t snapshot
 NAME                              USED  AVAIL  REFER  MOUNTPOINT
 tank/lxc/orig@snapshot               0      -   322M  -
 # lxc-destroy -n snapshot
 # zfs list -t snapshot
 NAME                              USED  AVAIL  REFER  MOUNTPOINT
 tank/lxc/orig@snapshot               0      -   322M  -
 # lxc-destroy -n orig
 cannot destroy 'tank/lxc/orig': filesystem has children
 use '-r' to destroy the following datasets:
 tank/lxc/orig@snapshot
 lxc_container: Error destroying rootfs for orig
 Destroying orig failed
 # lxc-create --version
 1.0.6
 # cat /etc/issue
 Ubuntu 14.04.1 LTS \n \l

**Comment by naisanza** [comment]
Experiencing the same thing

```
Linux lxc 4.4.0-15-generic #31-Ubuntu SMP Fri Mar 18 19:08:31 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

Snapshots created by lxc-copy are not destroyed when using lxc-destroy


```
root@lxc:~# lxc version
2.0.0.rc6
```

**Comment by brauner** [comment]
Should be fixed by #1184.
